libmuscle/python/libmuscle/manager/profile_store.py

```diff
@@ -74,16 +138,51 @@ def to_tuple(e: ProfileEvent) -> Record:
             return (
                     instance_oid, e.event_type.value, e.start_time.nanoseconds,
                     e.stop_time.nanoseconds, port_name, port_operator,
-                    e.port_length, e.slot, e.message_size, e.message_timestamp)
+                    e.port_length, e.slot, e.message_number, e.message_size,
+                    e.message_timestamp)
+
+        cur = self._get_cursor()
+        batch = self._queue.get()
+        while batch is not None:
+            instance_id, events = batch
+
+            cur = self._get_cursor()
+            cur.execute("BEGIN IMMEDIATE TRANSACTION")
+
+            instance_oid = self._get_instance_oid(cur, instance_id)
+
+            cur.executemany(
+                    "INSERT INTO events"
+                    " (instance_oid, event_type_oid, start_time, stop_time,"
+                    "  port_name, port_operator_oid, port_length, slot,"
+                    "  message_number, message_size, message_timestamp)"
+                    " VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                    map(to_tuple, events))
+            cur.execute("COMMIT")
+            if _SYNCHED:
+                self._confirmation_queue.put(None)
+            batch = self._queue.get()
 
-        cur.executemany(
-                "INSERT INTO events"
-                " (instance, event_type, start_time, stop_time, port_name,"
-                "  port_operator, port_length, slot, message_size,"
-                "  message_timestamp) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
-                map(to_tuple, events))
-        cur.execute("COMMIT")
         cur.close()
+        super().close()
+
+    def _get_instance_oid(self, cur: Cursor, instance_id: Reference) -> int:
+        """Get the oid for a given instance.
+
+        Inserts instance into the database and returns the new oid if
+        needed. Does not start its own transaction.
```

Comment:
I don't see an INSERT statement in this method?

Reply:
Ah, this changed and the comment is obsolete. Will fix.